docs: Update the documentation section for the IntelliJ-based IDEs (#2104)
codenvy/che-docs: modules/end-user-guide/partials/proc_provisioning-jetbrains-activation-code-for-offline-use.adoc

[id="provisioning-jetbrains-activation-code-for-offline-use_{context}"]

> = Provisioning JetBrains activation code for offline use

Reviewer: ```suggestion
= Provisioning the JetBrains offline activation code
```

> This section describes how to provision the JetBrains activation code to {prod-short}, when using link:https://www.jetbrains.com/store/[a JetBrains IDE requesting a registration]. After each subscription renewal, you need to generate an activation code and apply it to your product.

Reviewer: ```suggestion
Some editions of JetBrains IDEs require a paid subscription beyond the evaluation period, which means buying a license from JetBrains. To register a license, you need to provision to Che the JetBrains activation code for offline usage. When you renew your subscription, you will need to generate a new activation code and provision the new offline activation code.
```

Reviewer 2: _To register a license, you need to provision_ ->
_To register a license, provision_

_When you renew your subscription, you will need to generate_ ->
_When renewing the subscription, generate_

.Prerequisites

* An active link:https://account.jetbrains.com/login[JetBrains account].

> * An active link:https://www.jetbrains.com/store/comparison.html[personal or organization subscription].

Reviewer: ```suggestion
* An active link:https://sales.jetbrains.com/hc/en-gb[subscription for organizations or for individual use].
```

Reviewer: just aligning the terminology with https://www.jetbrains.com/store/#commercial?billing=yearly

Reviewer 3: OK but ket'sd keep the link to the www website.

```suggestion
* An active link:https://www.jetbrains.com/store/comparison.html[subscription for organizations or for personal use].
```

* A `base64` conversion tool.

> * An image containing the editor, see xref:building-images-for-ides-based-on-the-intellij-platform.adoc[].

Reviewer: ```suggestion
* An image containing the editor. See xref:building-images-for-ides-based-on-the-intellij-platform.adoc[].
```

Reviewer: I believe we need a period here. Otherwise, we're asking an image containing the editor to see the link.

> * A workspace running with the editor, see xref:using-intellij-based-ides.adoc[].

Reviewer: ```suggestion
* A workspace running with the editor. See xref:using-intellij-based-ides.adoc[].
```

Reviewer: Same thing.

.Procedure

> . Retrieve the activation code from the JetBrains account. JetBrains provides a zip archive with two types of the activation code. Use the `<License{nbsp}ID>{nbsp}-{nbsp}for{nbsp}2018.1{nbsp}or{nbsp}later.txt` file. Save the content in the `jetbrains_license.txt` file.

Reviewer: ```suggestion
. Generate the activation code by logging in to your link:https://account.jetbrains.com/login[JetBrains Account] and clicking the *Download activation code for offline usage* link.
. Save `<License{nbsp}ID>{nbsp}-{nbsp}for{nbsp}2018.1{nbsp}or{nbsp}later.txt` from the downloaded zip archive as `jetbrains_license.txt`.
```

> . To define __<base64_encoded_activation_code>__, calculate the base64 encoded version of the content of the activation code.

Reviewer: Is there a simpler way to describe this step than to`calculate the ...`?
For example,
```suggestion
. Run the `base64` command on `jetbrains_license.txt` to convert the activation code to base64 encoding for use as __<base64_encoded_activation_code>__ in the next step.
```

    app.kubernetes.io/component: workspace-secret
    app.kubernetes.io/part-of: che.eclipse.org
  annotations:
    che.eclipse.org/automount-workspace-secret: 'false' <1>
    che.eclipse.org/mount-path: /tmp/
    che.eclipse.org/mount-as: file
data:
  idea.key: __<base64_encoded_activation_code>__
  pycharm.key: __<base64_encoded_activation_code>__
  webstorm.key: __<base64_encoded_activation_code>__
  phpstorm.key: __<base64_encoded_activation_code>__
  goland.key: __<base64_encoded_activation_code>__
----
+
<1> `+che.eclipse.org/automount-workspace-secret: 'false'+`: disables the mounting process until a workspace component explicitly requests it with the `+automountWorkspaceSecrets: true+` property.

. Apply the {orch-name} Secret to the {orch-name} {orch-namespace} running the workspace.
+
[subs="+quotes,+attributes,+macros"]
----
$ {orch-cli} apply -f secret.yaml
----

. To mount the activation codes into a workspace, update the workspace devfile configuration to set `+automountWorkspaceSecrets: true+`.
+
[source,yaml,subs="+quotes,+attributes,+macros"]
----
components:
  - type: cheEditor
    automountWorkspaceSecrets: true
    reference: "__<URL_to_meta.yaml>__"
----

> . Restart the workspace

Reviewer: ```suggestion
. Restart the workspace.
```

[id="provisioning-jetbrains-activation-code-for-offline-use_{context}"]
= Provisioning the JetBrains offline activation code

> Some editions of JetBrains IDEs require a paid subscription beyond the evaluation period, which means buying a license from JetBrains. To register a license, you need to provision to Che the JetBrains activation code for offline usage. When you renew your subscription, you will need to generate a new activation code and provision the new offline activation code.

Reviewer: ```suggestion
Some editions of JetBrains IDEs require a paid subscription beyond the evaluation period, which means buying a license from JetBrains. To register a license, you need to provision to Che the JetBrains activation code for offline usage. When you renew your subscription, you will need to generate and provision a new offline activation code.
```

.Prerequisites

* An active link:https://account.jetbrains.com/login[JetBrains account].
* An active link:https://www.jetbrains.com/store/comparison.html[subscription for organizations or for personal use].
* A `base64` conversion tool.

> * An image containing the editor. See xref:building-images-for-ides-based-on-the-intellij-platform.adoc[].

Reviewer: ```suggestion
* An image containing the IDE. See xref:building-images-for-ides-based-on-the-intellij-platform.adoc[].
```

> * A workspace running with the editor. See xref:using-intellij-based-ides.adoc[].

Reviewer: ```suggestion
* A workspace running with the IDE. See xref:using-intellij-based-ides.adoc[].
```

.Procedure

> . Generate the activation code by logging in to your link:https://account.jetbrains.com/login[JetBrains Account] and click the *Download activation code for offline usage* link.

Reviewer: ```suggestion
. Generate the activation code by logging in to your link:https://account.jetbrains.com/login[JetBrains Account] and clicking the *Download activation code for offline usage* link.
```

Reviewer 3: clicking generates a vale error

Reviewer 3: Rewritten, completely.

. Save `<License{nbsp}ID>{nbsp}-{nbsp}for{nbsp}2018.1{nbsp}or{nbsp}later.txt` from the downloaded zip archive as `jetbrains_license.txt`.

> . Run the `base64` command on `jetbrains_license.txt` to convert the activation code to base64 encoding for use as __<base64_encoded_activation_code>__ in the next step.

Reviewer: ```suggestion
. Run the `base64` command on `jetbrains_license.txt` to convert the activation code to base64 encoding for use as `__<base64_encoded_activation_code>__` in the next step.
```

Reviewer 3: done. 
Use openssl base64 as it has a consistent implementation on various platforms. (as opposed to the  `base64` tool)
